packages/langium/src/lsp/completion/completion-provider.ts

```diff
@@ -381,28 +376,26 @@ export class DefaultCompletionProvider implements CompletionProvider {
         const assignment = getContainerOfType(crossRef.feature, ast.isAssignment);
         let node = context.node;
         if (assignment && node) {
-            if (crossRef.type && (crossRef.new || node.$type !== crossRef.type)) {
+            if (crossRef.type) {
                 node = {
```

Comment:
why is `node` re-initialized although it is already truthy?
Is this branch supposed to cover the case of assigning a (the first) property of an astNode not existing in the AST yet?
Under which circumstances is `crossRef.type` (un-)initialized?
The var name `crossRef` is a bit misleading as it actually denotes a `NextFeature` referring to a crossRef. Esp. `crossRef.type` is confusing, as it denotes `NextFeature.type`, not `CrossReference.type`.

Please add some comments here (and there).

Reply:
We sometimes have parser rules that look like this:

```
Ref: ref=[Element:ID];
```

Since we want the next element in our completion algorithm, we actually complete the `ref` assignment/cross reference although the current element cannot be of type `Ref`. In this case, the `NextFeature#type` property is set to `Ref` (since it's the new containing type), and we build a new synthetic element that can be used to compute the cross reference.

It's generally uninitialized when we don't enter a new parser rule during completion. I.e. like in the hello world grammar:

```
Hello |
```

Where the cross reference at `|` is part of the same parser rule as `Hello`.

Reply:
Thanks a lot 👍